### PR TITLE
[wip] Bug 1711533: Remove discovery access from system:unauthenticated

### PR DESCRIPTION
**Pull request by enj**

### CODE REVIEW

File: test/extended/authorization/rbac/groups_default_rules.go [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+		e2e.Failf("%s has extra cluster wide permissions:\n%s", group, rulesToSting(missing))
+	}
+
+	// force test data to be cleaned up every so often but allow extra rules to not deadlock new changes
+	if cover, missing := validation.Covers(actualRules, expectedRules); !cover && len(missing) > 5 {
+		e2e.Failf("test data for %s has too many unnecessary permissions:\n%s", group, rulesToSting(missing))
+	}
+}
+
+func rulesToSting(rules []rbacv1.PolicyRule) string {
```
Comment (ericavonb) [inline test/extended/authorization/rbac/groups_default_rules.go]: typo: `s/Sting/String/`

**Comment by abhinavdahiya** [comment]
Cross linking a requirement before this merges:
https://github.com/openshift/installer/pull/1835#issuecomment-501774669
https://github.com/openshift/installer/pull/1835#issuecomment-501775343

/cc @smarterclayton

File: pkg/cmd/server/bootstrappolicy/policy.go [hunk 2]
```diff
@@ -102,14 +102,9 @@ var (
...
 			"/api", "/api/*",
 			"/apis", "/apis/*",
-			"/oapi", "/oapi/*",
 			"/openapi/v2",
 			"/swaggerapi", "/swaggerapi/*", "/swagger.json", "/swagger-2.0.0.pb-v1",
-			"/osapi", "/osapi/", // these cannot be removed until we can drop support for pre 3.1 clients
 			"/.well-known", "/.well-known/*",
-
-			// we intentionally allow all to here
-			"/",
```
Comment (adambkaplan) [inline pkg/cmd/server/bootstrappolicy/policy.go]: Ditto.

**Comment by adambkaplan** [comment]
fyi @gabemontero @akram

**Comment by adambkaplan** [comment]
@enj fix on the Jenkins side is making its way to OpenShift. See https://github.com/fabric8io/kubernetes-client/pull/1642

Per prior discussion starting discovery on the `/apis/` endpoint is the right approach, correct?
fyi @akram @waveywaves @gabemontero

**Comment by akram** [comment]
PR https://github.com/openshift/jenkins-sync-plugin/pull/320 is pending on the Jenkins Sync Plugin to also allow this.

**Comment by enj** [comment]
@mfojtik @sttts @stlaz you may care one day.

/close

**Pull request closed without merging**